### PR TITLE
Structured project folders and started /joints routing

### PR DESCRIPTION
**Pull request by gracewgao**
`/joints`
`/joints/<joint_id>`
`/joints/<joint_id>/menu`
^These should return the basic data from the database in json format

### CODE REVIEW

File: src/webserver/flick/__init__.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+   app = Flask(__name__, instance_relative_config=True)
+
+   @app.route('/')
+   def hello_world():
+      return 'Hello World'
+
+   if __name__ == '__main__':
+      app.run(debug=True)
+
+   from . import db
```
Comment (akshgpt7) [inline src/webserver/flick/__init__.py:15]: Make sure all the imports are at the top of the file.

File: src/webserver/flick/__init__.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+   db.init_app(app)
+
+   from . import joints
```
Comment (akshgpt7) [inline src/webserver/flick/__init__.py:18]: Import at top of the file.

File: src/webserver/flick/joints.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+from werkzeug.exceptions import abort
+from .db import get_db
+
+bp = Blueprint('joints', __name__)
+
+@bp.route('/joints')
+def get_joints():
+    db = get_db()
+    joints = db.execute(
+        'SELECT joint_id, name, location, description FROM joints'
```
Comment (akshgpt7) [inline src/webserver/flick/joints.py:14]: You can simply write `SELECT * FROM joints` to select all the fields.
Please make sure to change it in all queries.

**Review by akshgpt7 — CHANGES_REQUESTED** [review]
Thanks for this! It's looking great.

Just a few general review comments, apart from the specific ones:
- I can't figure out how to run the server, which file do I have to run?
- The database import is giving me an error. Did you try running it locally?

File: src/webserver/flick/__init__.py (hunk 1 above)
Comment (gracewgao) [reply src/webserver/flick/__init__.py:15]: I'll change that, thanks!

Comment (akshgpt7) [reply src/webserver/flick/__init__.py:15]: Ah, I realized that they need to be within the flask app scope, so you can leave it this way.

**Pull request merged**